ext/opentelemetry-ext-datadog/src/opentelemetry/ext/datadog/__init__.py

```diff
@@ -0,0 +1,211 @@
+                    https=url_parsed.scheme == "https",
+                )
+            elif url_parsed.scheme == "unix":
+                self._agent_writer = AgentWriter(uds_path=url_parsed.path)
+            else:
+                raise ValueError(
+                    "Unknown scheme `%s` for agent URL" % url_parsed.scheme
+                )
+        return self._agent_writer
+
+    def export(self, spans):
+        datadog_spans = self._translate_to_datadog(spans)
+
+        self.agent_writer.write(spans=datadog_spans)
+
+        return SpanExportResult.SUCCESS
+
+    def shutdown(self):
+        if self.agent_writer.started:
+            self.agent_writer.stop()
+            self.agent_writer.join(self.agent_writer.exit_timeout)
+
+    def _translate_to_datadog(self, spans):
+        datadog_spans = []
+
+        for span in spans:
+            trace_id, parent_id, span_id = _get_trace_ids(span)
+
+            # datadog Span is initialized with a reference to the tracer which is
+            # used to record the span when it is finished. We can skip ignore this
+            # because we are not calling the finish method and explictly set the
+            # duration.
+            tracer = None
+
+            datadog_span = DatadogSpan(
+                tracer,
+                span.name,
+                service=self.service,
+                resource=_get_resource(span),
+                span_type=_get_span_type(span),
+                trace_id=trace_id,
+                span_id=span_id,
+                parent_id=parent_id,
+            )
+            datadog_span.start_ns = span.start_time
+            datadog_span.duration_ns = span.end_time - span.start_time
+
+            if span.status.canonical_code is not StatusCanonicalCode.OK:
+                datadog_span.error = 1
+                if span.status.description:
+                    exc_type, exc_val = _get_exc_info(span)
+                    # no mapping for error.stack since traceback not recorded
+                    datadog_span.set_tag("error.msg", exc_val)
+                    datadog_span.set_tag("error.type", exc_type)
+
+            datadog_span.set_tags(span.attributes)
+
+            datadog_spans.append(datadog_span)
+
+        return datadog_spans
+
+
+def _get_trace_ids(span):
+    """Extract tracer ids from span"""
+    ctx = span.get_context()
+    trace_id = ctx.trace_id
+    span_id = ctx.span_id
+
+    if isinstance(span.parent, trace_api.Span):
+        parent_id = span.parent.get_context().span_id
+    elif isinstance(span.parent, trace_api.SpanContext):
+        parent_id = span.parent.span_id
+    else:
+        parent_id = 0
+
+    trace_id = _convert_trace_id_uint64(trace_id)
+
+    return trace_id, parent_id, span_id
+
+
+def _convert_trace_id_uint64(otel_id):
+    """Convert 128-bit int used for trace_id to 64-bit unsigned int"""
+    raw = otel_id.to_bytes(16, "big")
+    return int.from_bytes(raw[8:], byteorder="big")
```

Comment:
I think you can just use `otel_id & 0xFFFFFFFFFFFFFFFF` to get the lower 16 bits.